src/main/java/org/jenkinsci/plugins/github_branch_source/DefaultGitHubNotificationStrategy.java

```diff
@@ -51,8 +51,7 @@
      */
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        return (o == null || getClass() != o.getClass());
+        return o instanceof DefaultGitHubNotificationStrategy;
```

Comment:
So this is a landmine if you ever remove the `final` from the class. I recommend either leaving a comment to update if the class is made non-final, or fixing the bug... namely it should be

```
return this == o || (o != null && getClass == o.getClass());
```

or if you want the minimal diff on your previous code:

```
if (this == 0) return true;
return !(o == null || getClass() != o.getClass());
```